### PR TITLE
adds dns prefetching for most required origins

### PR DESCRIPTION
**Pull request by mdole**
Took a first stab at adding DNS prefetching with a naive approach—prefetch everything always. Questions to resolve:

1. Is this correctly implemented + having an impact? I'm not sure how best to test this since locally hosted force seems to fetch less/different things. @zephraph any tips you have here would be appreciated
2. We don't need to do every one of these for every page (see screenshots below). How can we make sure that these resources are only being prefetched when they're actually used? I could possibly run Lighthouse on most major page types and see what it recommends prefetching on each

Tracked [in Jira](https://artsyproduct.atlassian.net/browse/PLATFORM-1196)

From a Lighthouse run on a production artwork page, https://www.artsy.net/artwork/nils-udo-composition-aux-fruits (mobile 3g, default settings):
screen shot 2019-02-20 at 10 33 50 pm

From a Lighthouse run on the production /auctions/ page (same settings):
screen shot 2019-02-20 at 10 44 55 pm

### CODE REVIEW

**Comment by mdole** [comment]
gentle nudge @zephraph—would appreciate your thoughts when you have a moment!

File: src/desktop/components/main_layout/templates/head.jade [hunk 1]
```diff
@@ -5,6 +5,24 @@ link( rel='apple-touch-icon', sizes='152x152', href=asset('/images/icon-152.png'
...
+link( rel='dns-prefetch', href=asset('https://volley.artsy.net') )
+link( rel='dns-prefetch', href=asset('https://api.artsy.net') )
+link( rel='dns-prefetch', href=asset('https://webfonts.artsy.net') )
+link( rel='dns-prefetch', href=asset('https://connect.facebook.net') )
+link( rel='dns-prefetch', href=asset('https://www.googleadservices.com') )
+link( rel='dns-prefetch', href=asset('https://cdn.segment.com') )
+link( rel='dns-prefetch', href=asset('https://googleads.g.doubleclick.net') )
+link( rel='dns-prefetch', href=asset('https://s.adroll.com') )
+link( rel='dns-prefetch', href=asset('https://www.google-analytics.com') )
+link( rel='dns-prefetch', href=asset('https://sb.scorecardresearch.com') )
```
Comment (zephraph) [inline src/desktop/components/main_layout/templates/head.jade]: You don't need to use the `asset` method for these. Being as they're full urls you should just be able to do `href='https://www.google-analytics.com'`.

Comment (zephraph) [reply src/desktop/components/main_layout/templates/head.jade]: I think this list is likely too exhaustive and will load some unnecessary resources. Try to focus on the results of new pages... artist, artwork, collect.

**Comment by zephraph** [comment]
Hey @mdole, when you get a chance, could you take this PR out of the draft state? I'll push it along after that. Thanks!

**Comment by mdole** [comment]
Thanks for the updates @zephraph and apologies for leaving it stuck in draft state—didn't know that PRs in draft mode couldn't be marked as ready except by the author.

Since you're OOO this week and have approved, I'm going to go ahead and merge this one.

**Pull request merged**